src/matter/cluster/server/OnOffServer.ts

```diff
@@ -0,0 +1,160 @@
+    //      I know that it is the wrong place in the "Handlers" object, but I don't know where to put it right now
+    //      because it should be "bound to the cluster instance" and not to the "cluster type".
+    //      Yes could also be build without the subscriptions, but so the code is separated from the other implementation
+    //      parts and if used as library and the "specific device implementation" manipulates onTime or offWaitTime then
+    //      it is all handled automatically and pot. not missed.
+    constructor: ({ attributes: { onOff, onTime, offWaitTime, globalSceneControl } }) => {
+        this.onTimeTimeout: NodeJS.Timeout | null = null;
+        onTime.addMatterListener((value) => {
+            if (this.onTimeTimeout) {
+                clearTimeout(this.onTimeTimeout);
+                this.onTimeTimeout = null;
+            }
+            if (value === 0) {
+                offWaitTime.set(0);
+                onOff.set(false);
+            }
+            if (value > 0 && onOff.get()) {
+                // TODO This is the 100% accurate impelementation right now ...
+                //      maybe could be optimized without a "100ms timeout loop",
+                //      but for now should do exactly what specs say
+                this.onTimeTimeout = setTimeout(() => {
+                    this.onTimeTimeout = null;
+                    onTime.set(value - 1);
+                }, 100);
+            }
+        });
+
+        this.offWaitTimeout: NodeJS.Timeout | null = null;
+        offWaitTime.addMatterListener((value) => {
+            if (this.offWaitTimeout) {
+                clearTimeout(this.offWaitTimeout);
+                this.offWaitTimeout = null;
+            }
+            if (value > 0 && !onOff.get()) {
+                this.offWaitTimeout = setTimeout(() => {
+                    this.offWaitTimeout = null;
+                    offWaitTime.set(value - 1);
+                }, 100);
+            }
+        });
+    },
+    // TODO The trick that we used in Device.ts to access "optionalAttributes" ... how to do it here?
+    //      In fact I need to get globalSceneControl
+    on: async ({ attributes: { onOff, onTime, offWaitTime, globalSceneControl } }) => {
+        onOff.set(true);
+        globalSceneControl.set(true);
+        if (onTime.get() === 0) {
+            offWaitTime.set(0);
+        }
+    },
+    off: async ({ attributes: { onOff, onTime, offWaitTime } }) => {
+        onOff.set(false);
+        onTime.set(0);
+    },
+    toggle: async ({ attributes: { onOff, onTime, offWaitTime } }) => {
+        if (onOff.get()) {
+            onOff.set(false);
+            onTime.set(0);
+        } else {
+            onOff.set(true);
+            if (onTime.get() === 0) {
+                offWaitTime.set(0);
+            }
+        }
+    },
+    offWithEffect: async ({request: { effectId, effectVariant}, attributes: { onOff, globalSceneControl } }) => {
```

Comment:
Needs access to level Cluster (as soon as we implement it)

Reply:
Let's try if passing other clusters in the constructor would work.
At the end, the device (in the Matter definition) will be responsible for instantiating all cluster serves and orchestrating them so it can pass other clusters instances when starting up.

Reply:
In fact also "second implementation step". But already here this way will not work that good nbecause LevelControl cluster needs OnOff Cluster instance and OnOff CLuster needs Level Control cluster because they _both_ interact with each other. So Would only work if we modify config after instanciation :-)
Maybe a topic for the call tonight